py/torch_tensorrt/dynamo/conversion/_ConverterRegistry.py

```diff
@@ -324,13 +414,30 @@ def __getitem__(
 
                 if isinstance(converters, (list, tuple)):
                     for candidate in converters:
-                        if candidate.capability_validator(node):
+                        if candidate.capability_validator(node) and (
+                            self.assume_dynamic_shape_support
+                            or (
+                                has_dynamic_shapes(node)
+                                and candidate.supports_dynamic_shapes
+                            )
+                        ):
+                            # If node has dynamic inputs and the converter supports dynamic shapes, it is enabled
+                            return (
+                                candidate.converter_implementation,
+                                calling_convention,
+                            )
+                        elif candidate.capability_validator(
+                            node
+                        ) and not has_dynamic_shapes(node):
+                            # For static shapes all converters are turned on based on capability_validator check
```

Comment:
Consider condensing logic to:
```python
node_has_dynamic_shapes = has_dynamic_shapes(node)

if candidate.capability_validator(node) and (
    self.assume_dynamic_shape_support
    or not node_has_dynamic_shapes 
    or (
        node_has_dynamic_shapes 
        and candidate.supports_dynamic_shapes
    )
):
```
The above might assist with showing explicitly the three conditions that allow a converter to "pass", from my understanding:

- Validator Passes [Required]
- Need at least _one_ of the following:
  - `assume_dynamic_shape_support` [Optional]
  - Static Node [Optional]
  - Dynamic Node + Dynamic Support Converter [Optional]